auction/serializers.py

```diff
@@ -0,0 +1,143 @@
+from typing import Any, Dict
+from django.utils import timezone
+from rest_framework import serializers
+
+from auction.models import Auction
+from auction.models import AuctionImage
+from auction.models import Bid
+from user.models import User
+
+
+class UserSerializer(serializers.ModelSerializer):
+    """
+    Serializer for User model.
+    """
+    class Meta:
+        model = User
+        fields = ["username", "user_image"]
+
+
+class BidSerializer(serializers.ModelSerializer):
+    """
+    Serializer for Bid model.
+    """
+    user = UserSerializer(read_only=True)
+
+    class Meta:
+        model = Bid
+        fields = "__all__"
+
+    def create(self, validate_data: Dict[str, Any]) -> Bid:
+        """
+        Method to create a new bid.
+        """
+        user = self.context["request"].user
+        auction = validate_data["auction"]
+        bid_value = validate_data["bid"]
+
+        if not auction.bid:
+            if auction.start_bid <= bid_value:
+                bid = self.create_bid(user, validate_data, auction)
+            else:
+                raise serializers.ValidationError(
+                    "Bid value must be at least equal to the start bid."
+                )
+        elif (auction.bid + auction.bid_step) <= bid_value:
+            bid = self.create_bid(user, validate_data, auction)
+        else:
+            raise serializers.ValidationError(
+                "Bid value must be higher than the current bid plus the bid step."
+            )
+
+        return bid
+
+    def create_bid(self, user, validate_data: Dict[str, Any], auction: Auction) -> Bid:
+        bid = Bid.objects.create(user=user, **validate_data)
+        auction.bid = bid.bid
+        auction.save()
+        return bid
```

Comment:
that method must be higher than create method, because you use this method in create func